sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py

```diff
@@ -1,36 +1,117 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
+# pylint: disable=redefined-builtin
+from typing import Dict, Optional, Union, List
+from azure.ai.ml._restclient.v2020_09_01_dataplanepreview.models import (
+    BatchJobResource,
+    BatchLoggingLevel,
+    DataVersion,
+    InferenceDataInputBase,
+    JobInput,
+    JobOutputV2,
+    BatchJob as RestBatchJob,
+)
+from azure.ai.ml.entities._job.compute_configuration import ComputeConfiguration
+from azure.ai.ml.entities._deployment.deployment_settings import BatchRetrySettings
 
-from azure.ai.ml._restclient.v2020_09_01_dataplanepreview.models import BatchJobResource
-
-
+# pylint: disable=too-many-instance-attributes
 class BatchJob(object):
     """Batch jobs that are created with batch deployments/endpoints invocation.
 
     This class shouldn't be instantiated directly. Instead, it is used as the return type of batch deployment/endpoint
     invocation and job listing.
     """
 
-    def __init__(self, **kwargs):
-        self.id = kwargs.get("id", None)
-        self.name = kwargs.get("name", None)
-        self.type = kwargs.get("type", None)
-        self.status = kwargs.get("status", None)
-
-    def _to_dict(self):
-        return {
-            "id": self.id,
-            "name": self.name,
-            "type": self.type,
-            "status": self.status,
-        }
+    def __init__(
+        self,
+        *,
+        id: Optional[str] = None,
+        name: Optional[str] = None,
+        type: Optional[str] = None,
+        status: Optional[str] = None,
+        compute: Optional[ComputeConfiguration] = None,
+        dataset: Optional[InferenceDataInputBase] = None,
+        description: Optional[str] = None,
+        error_threshold: Optional[int] = None,
+        input_data: Optional[Dict[str, JobInput]] = None,
+        logging_level: Optional[Union[str, BatchLoggingLevel]] = None,
+        max_concurrency_per_instance: Optional[int] = None,
+        mini_batch_size: Optional[int] = None,
+        job_name: Optional[str] = None,
+        output_data: Optional[Dict[str, JobOutputV2]] = None,
+        output_dataset: Optional[DataVersion] = None,
+        output_file_name: Optional[str] = None,
+        partition_keys: Optional[List[str]] = None,
+        properties: Optional[Dict[str, str]] = None,
+        retry_settings: Optional[BatchRetrySettings] = None,
+        tags: Optional[Dict[str, str]] = None,
+        **kwargs  # pylint:disable=unused-argument
+    ):
```

Comment:
@TajiHarrisMicrosoft: Please add SystemData here as well. Removing it would be a breaking change.